crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -655,6 +685,47 @@ impl<'a, 'o> TimelineEventHandler<'a, 'o> {
         }
     }
 
+    /// Handle a beacon_info event and update the `BeaconInfoState`.
+    ///
+    /// TODO: Handle case of `m.beacon_info` event that is identical to the first, except with live: false.
+    /// This stops the live location share.
+    ///
+    /// TODO: How do you search for an existing beacon_info event in the timeline by state key?
+    fn handle_beacon_info(
+        &mut self,
+        c: FullStateEventContent<BeaconInfoEventContent>,
+        user_id: OwnedUserId,
+    ) {
+        let mut beacon_state = BeaconState::new(c, user_id.clone());
```

Comment:
Use case of an m.beacon_info event that is identical to the first, except with live: false..

How do you search for an existing `beacon_info` event in the timeline by `state_key` in order to update the `beacon_info_state`?